code/modules/projectiles/guns/syringe_gun.dm

```diff
@@ -95,3 +94,285 @@
+/obj/item/gun/syringe/rapidsyringe/proc/insert_single_syringe(obj/item/reagent_containers/syringe/new_syringe, mob/user)
+	if(new_syringe.reagents.total_volume)
+		if(user)
+			to_chat(user, "<span class='warning'>[src] only accepts empty syringes.</span>")
+		return FALSE
+	var/in_clip = length(syringes) + (chambered?.BB ? 1 : 0)
+	if(in_clip >= max_syringes)
+		if(user)
+			to_chat(user, "<span class='warning'>[src] is full!</span>")
+		return FALSE
+
+	if(user)
+		if(!user.unEquip(new_syringe))
+			return
+		to_chat(user, "<span class='notice'>You load \the [new_syringe] into [src].</span>")
+		playsound(src, 'sound/weapons/gun_interactions/bulletinsert.ogg', 50, 1)
+	syringes.Add(new_syringe)
+	new_syringe.forceMove(src)
+
+	process_chamber() // Chamber the syringe if none is already
+	return TRUE
+
+/obj/item/gun/syringe/rapidsyringe/attackby(obj/item/A, mob/user, params, show_msg)
+
+	if(istype(A, /obj/item/storage))
+		// Boxes can be dumped in.
+		var/obj/item/storage/container = A
+		if(!length(container.contents))
+			to_chat(user, "<span class='warning'>[A] is empty!</span>")
+			return TRUE
+		if(length(syringes) + (chambered?.BB ? 1 : 0) == max_syringes)
+			to_chat(user, "<span class='warning'>[src] is full!</span>")
+			return TRUE
+
+		var/total_inserted = 0
+		for(var/obj/item/reagent_containers/syringe/S in container)
+			if((length(syringes) + (chambered?.BB ? 1 : 0)) == max_syringes)
+				// full
+				break
+
+			// only allow empty syringes
+			if(!S.reagents.total_volume)
+				syringes += S
+				S.forceMove(src)
+				total_inserted++
+
+		if(total_inserted)
+			playsound(src, 'sound/weapons/gun_interactions/bulletinsert.ogg', 50, 1)
+			to_chat(user, "<span class='notice'>You load [total_inserted] empty syringes into [src].")
+			// Chamber a syringe.
+			process_chamber()
+
+	else if(istype(A, /obj/item/reagent_containers/syringe))
+		insert_single_syringe(A, user)
+
+	else if(istype(A, /obj/item/reagent_containers))
+		// Loading with chemicals (but not from syringes)
+		var/obj/item/reagent_containers/incoming = A
+		if(!incoming.is_drainable())
+			return
+
+		if(!incoming.reagents.total_volume)
+			to_chat(user, "<span class='warning'>[incoming] is empty!</span>")
+			return
+
+		if(reagents.holder_full())
+			to_chat(user, "<span class='warning'>[src]'s internal reservoir is full!</span>")
+			return
+
+		var/trans = incoming.reagents.trans_to(src, incoming.amount_per_transfer_from_this)
+		to_chat(user, "<span class='notice'>You transfer [trans] unit\s of the solution to [src]'s internal reservoir.</span>")
+		update_loaded_syringe()
+
+		// Reset the reservoir alarm
+		alarmed = FALSE
+
+	else
+		return ..()
+
+// Allow for emptying your deathmix, or for sec to find out what you were dumping into people
+/obj/item/gun/syringe/rapidsyringe/afterattack(atom/target, mob/living/user, flag, params)
+	if(istype(target, /obj/item/reagent_containers) && !istype(target, /obj/item/reagent_containers/syringe))
```

Comment:
Fairly sure this is uneeded because syringes fail the `!destination.is_refillable()` check below.
```suggestion
	if(istype(target, /obj/item/reagent_containers))
```

Reply:
Oh yeah, true. I think some of this is legacy from before I delegated it to container code.